### PR TITLE
Add dataResidencyRegion option for sending to portal

### PR DESCRIPTION
**Pull request by AnatolyPristensky**
* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [x] Did you check UI tests on the sample app? They are not executed on CI.

## Description

Add new API that allow to specify the data residency region for the current device.

## Related PRs or issues
[AB#102464](https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/102464)

### CODE REVIEW

File: AppCenter/AppCenter/MSACAppCenter.h [hunk 1]
```diff
@@ -109,6 +109,11 @@ NS_SWIFT_NAME(AppCenter)
  */
 @property(class, nonatomic, strong) NSString *logUrl;
 
+/**
+ * Data residency identifier.
```
Comment (DmitriyKirakosyan) [inline AppCenter/AppCenter/MSACAppCenter.h:113]:
```suggestion
 * Data residency region.
```

File: AppCenterCrashes/AppCenterCrashes/MSACCrashes.mm [hunk 2]
```diff
@@ -1339,7 +1339,7 @@ - (void)handleUserConfirmation:(MSACUserConfirmation)userConfirmation {
 
     // Second, get correlated user Id.
     log.userId = [[MSACUserIdContext sharedInstance] userIdAt:log.timestamp];
-
+
```
Comment (DmitriyKirakosyan) [inline AppCenterCrashes/AppCenterCrashes/MSACCrashes.mm]: remove

File: Sasquatch/Sasquatch.xcodeproj/xcshareddata/xcschemes/SasquatchWatchObjC.xcscheme [hunk 3]
```diff
@@ -41,8 +41,6 @@
       selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
       selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
       shouldUseLaunchSchemeArgsEnv = "YES">
-      <Testables>
-      </Testables>
       <MacroExpansion>
```
Comment (DmitriyKirakosyan) [inline Sasquatch/Sasquatch.xcodeproj/xcshareddata/xcschemes/SasquatchWatchObjC.xcscheme:44]: remove changes in watch projects.

File: Sasquatch/SasquatchSwift/AppDelegate.swift [hunk 4]
```diff
@@ -125,6 +125,12 @@ class AppDelegate: UIResponder, UIApplicationDelegate, CrashesDelegate, CLLocati
     if userId != nil {
       AppCenter.userId = userId;
     }
+    
+    // Set data residency region
+    let dataResidencyRegion = UserDefaults.standard.string(forKey: kMSACDataResidencyRegion)
+    if dataResidencyRegion != nil {
+      AppCenter.dataResidencyRegion = userId;
```
Comment (DmitriyKirakosyan) [inline Sasquatch/SasquatchSwift/AppDelegate.swift:132]: ?

File: SasquatchMac/SasquatchMac.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -924,7 +924,7 @@
 					};
 					D3C31C211E8E75B300B288CE = {
 						CreatedOnToolsVersion = 8.2.1;
-						DevelopmentTeam = UBF8T346G9;
+						DevelopmentTeam = 285W24646W;
```
Comment (DmitriyKirakosyan) [inline SasquatchMac/SasquatchMac.xcodeproj/project.pbxproj:927]: revert these changes

File: SasquatchMac/SasquatchMac/AppCenterDelegate.swift [hunk 6]
```diff
@@ -18,6 +18,7 @@ import AppCenter
   func isDebuggerAttached() -> Bool
   func startAnalyticsFromLibrary()
   func setUserId(_ userId: String?)
+  func setDataResidencyRegion(_ userId: String?)
```
Comment (DmitriyKirakosyan) [inline SasquatchMac/SasquatchMac/AppCenterDelegate.swift:21]: ?

**Review by DmitriyKirakosyan — CHANGES_REQUESTED** [review]

File: AppCenter/AppCenterTests/MSACAppCenterTests.m [hunk 7]
```diff
@@ -1065,4 +1065,13 @@ - (void)testDisableNetworkRequestsAfterStart {
   XCTAssertEqual(tokenObject, self.channelGroupMock);
 }
 
+- (void)testSetDataResidencyRegion {
+  // If
+  NSString *expectedDataResidencyRegion = @"rg";
+  [MSACAppCenter setDataResidencyRegion:expectedDataResidencyRegion];
+
+  // Then
+  XCTAssertEqualObjects(expectedDataResidencyRegion, [MSACAppCenter sharedInstance].dataResidencyRegion);
```
Comment (MikhailSuendukov) [inline AppCenter/AppCenterTests/MSACAppCenterTests.m:1074]: I think you should use a static getter here instead of directly getting the property from the instance.

File: AppCenter/AppCenter/MSACAppCenter.h [hunk 8]
```diff
@@ -109,6 +109,11 @@ NS_SWIFT_NAME(AppCenter)
  */
 @property(class, nonatomic, strong) NSString *logUrl;
 
+/**
+ * Data residency region.
```
Comment (MikhailSuendukov) [inline AppCenter/AppCenter/MSACAppCenter.h:113]: Here you need to expand the annotation, as stated in the description of the work item.

File: AppCenter/AppCenterTests/MSACAppCenterTests.m (hunk 7 above)
Comment (AnatolyPristensky) [reply AppCenter/AppCenterTests/MSACAppCenterTests.m]: Corrected

File: AppCenter/AppCenter/MSACAppCenter.h (hunk 8 above)
Comment (AnatolyPristensky) [reply AppCenter/AppCenter/MSACAppCenter.h]: Corrected

File: AppCenter/AppCenter/Internals/MSACAppCenterInternal.h [hunk 9]
```diff
@@ -110,6 +112,13 @@ static NSString *const kMSACTrueEnvironmentString = @"1";
  */
 - (NSString *)logUrl;
 
+/**
+ * Get the data residency region.
+ *
+ * @return data residency region.
```
Comment (DmitriyKirakosyan) [inline AppCenter/AppCenter/Internals/MSACAppCenterInternal.h:118]: remove return

File: AppCenter/AppCenter/MSACAppCenter.h [hunk 10]
```diff
@@ -109,6 +109,11 @@ NS_SWIFT_NAME(AppCenter)
  */
 @property(class, nonatomic, strong) NSString *logUrl;
 
+/**
+ * Data residency region. Verify list of supported regions on <link>. Value outside of supported range is treated by backend as ANY.
```
Comment (DmitriyKirakosyan) [inline AppCenter/AppCenter/MSACAppCenter.h:113]: "Verify list.." from the next line

File: AppCenter/AppCenterTests/MSACAbstractLogTests.m [hunk 11]
```diff
@@ -159,6 +163,14 @@ - (void)testIsEqual {
 
   // Then
   XCTAssertFalse([self.sut isEqual:log]);
+
+  // When 
+  self.sut.userId = @"FAKE-USER-ID";
```
Comment (DmitriyKirakosyan) [inline AppCenter/AppCenterTests/MSACAbstractLogTests.m:168]: remove userid

Comment (AnatolyPristensky) [reply AppCenter/AppCenterTests/MSACAbstractLogTests.m:168]: Leave as is, because this value reverts previous test value.

**Review by DmitriyKirakosyan — APPROVED** [review]
